ddev/src/ddev/utils/scripts/check_pr.py

```diff
@@ -0,0 +1,189 @@
+"""
+Running this script by itself must not use any external dependencies.
+"""
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from __future__ import annotations
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+from typing import Iterator
+
+
+def requires_changelog(target: str, files: Iterator[str]) -> bool:
+    if target == 'ddev':
+        source = 'src/ddev/'
+    else:
+        if target == 'datadog_checks_base':
+            directory = 'base'
+        elif target == 'datadog_checks_dev':
+            directory = 'dev'
+        elif target == 'datadog_checks_downloader':
+            directory = 'downloader'
+        else:
+            directory = target.replace('-', '_')
+
+        source = f'datadog_checks/{directory}/'
+
+    return any(f.startswith(source) or f == 'pyproject.toml' for f in files)
+
+
+def git(*args) -> str:
+    try:
+        process = subprocess.run(
+            ['git', *args], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, encoding='utf-8', check=True
+        )
+    except subprocess.CalledProcessError as e:
+        raise OSError(f'{str(e)[:-1]}:\n{e.output}') from None
+
+    return process.stdout
+
+
+def get_added_lines(git_diff: str) -> list[str]:
+    files: dict[str, dict[int, str]] = {}
+    for modification in re.split(r'^diff --git ', git_diff, flags=re.MULTILINE):
+        if not modification:
+            continue
+
+        # a/file b/file
+        # new file mode 100644
+        # index 0000000000..089fd64579
+        # --- a/file
+        # +++ b/file
+        metadata, *blocks = re.split(r'^@@ ', modification, flags=re.MULTILINE)
+        before, after = metadata.strip().splitlines()[-2:]
```

Comment:
```suggestion
        *_, before, after = metadata.strip().splitlines()
```

Nit suggestion, less indexing is always better imho